src/pages/Profile/components/EditProfile.tsx

```diff
@@ -0,0 +1,155 @@
+import * as React from "react";
+import { FlexDiv } from "../../../react-design-system/FlexDiv";
+import { Text } from "../../../react-design-system/Text";
+import { Input } from "../../../react-design-system/Input";
+import Button from "../../../react-design-system/Button";
+import { useFormik } from "formik";
+import * as yup from "yup";
+import { IUserRegister } from "../../../interfaces/pages/Auth";
+import styled from "styled-components";
+
+const InputField = styled(Input)`
+  border: 2px solid lightgray;
+  border-radius: 10px;
+  width: 300px;
+`;
+const StyledButton = styled(Button)`
+  color: blueviolet;
+  border: 2px solid lightgray;
+  border-radius: 10;
+  float: right;
+  width: 100;
+`;
+
+const ErrorContainer = styled.div`
+  color: "red";
+  font-size: 10px;
+`;
+const EditProfile = (props: {
+  profile: IUserRegister;
+  updateProfile: (values: IUserRegister) => void;
+}) => {
+  const { name, phone, email, city, state, country, address } = props?.profile;
+
+  let schema = yup.object().shape({
+    name: yup.string().required("Full name is required"),
```

Comment:
I think yup was a great find by u

Reply:
yeah yup resolve a lot of problems for us.